statsmodels/tsa/exponential_smoothing/ets.py

```diff
@@ -1160,13 +1160,13 @@ def _loglike_internal(
         res = self._residuals(yhat, data=data)
         logL = -self.nobs / 2 * (np.log(2 * np.pi * np.mean(res ** 2)) + 1)
         if self.error == "mul":
-            # GH-7331: in some cases, yhat can become negative, so that a
-            # multiplicative model is no longer well-defined. To avoid these
-            # parameterizations, we clip negative values to very small positive
-            # values so that the log-transformation yields very large negative
-            # values.
-            yhat[yhat <= 0] = 1 / (1e-8 * (1 + np.abs(yhat[yhat <= 0])))
-            logL -= np.sum(np.log(yhat))
+            # In some cases, yhat can become negative or zero, so that a
+            # multiplicative model is no longer well-defined. Zero values
+            # are replaced with 10^-32 (a very small number) and we take
+            # the absolute of yhat to avoid computing the log of negative
+            # numbers.
+            yhat[yhat == 0] = 1e-32
+            logL -= np.sum(np.log(np.abs(yhat)))
```

Comment:
I don't think reflecting negatives is a good idea.  Better to replace with small numbers.

Reply:
Please note that the absolute value is not there to avoid negative numbers, but instead arises mathematically:

If $\epsilon_t \sim \mathcal{N}(0, \sigma^2)$ then $y_t = \mu_{y,t}(1+\epsilon_t) \sim \mathcal{N}(\mu_{y,t}, \mu_{y,t}^2 \sigma^2)$. We then end up with the following log-likelihood: 

$\ell(\boldsymbol{\theta}, {\sigma}^2 | \mathbf{y}) = -\frac{1}{2} \sum_{t=1}^T \log(2 \pi \mu_{y,t}^2 \sigma^2) -\frac{1}{2} \sum_{t=1}^T \frac{\epsilon_t^2}{\sigma^2}$ . As you see, we have $\mu_{y,t}^2$ in the formula above, which after simplifications can be written as: $-\frac{T}{2} \log(2 \pi \sigma^2) -\frac{1}{2} \sum_{t=1}^T \frac{\epsilon_t^2}{\sigma^2} - \sum_{t=1}^T \log |\mu_{y,t}|$.

So, it is necessary to have it there for the correct calculation of the log-likelihood. The derivations are based on the Section 11.1 here: https://openforecast.org/adam/ADAMETSEstimationLikelihood.html

Reply:
Thanks for reviewing the PR @bashtage and thanks for the additional explanation on the likelihood derivation @config-i1. 

Based on @config-i1's explanation and the other implementations, I think it makes sense to keep the `np.abs(yhat)` do you agree @bashtage?